### PR TITLE
📖 Documentation for go/v2 and go/v3 plugin

### PR DESCRIPTION
**Pull request by Ashwin901**
As mentioned in the issue #2542 I have added the documentation for the go/v2, go/v3 and also the available plugins page.
I have tried my best to follow all the checkpoints mentioned in the issue, but please let me know if any changes are required.

### CODE REVIEW

File: docs/book/src/plugins/available-plugins.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Available plugins
+
+This section describes the plugins supported and shipped in with the Kubebuilder project see: 
+- [go/v3](go-v3-plugin.md)
+- [go/v2](go-v2-plugin.md)
+<!-- Plugins are configured to execute one or more subcommands. The main subcommands are:
+- Init
+- Edit
+- Create API
+- Create Webhook -->
```
Comment (camilamacedo86) [inline docs/book/src/plugins/available-plugins.md:10]: A plugin can have 1 or many interfaces implemented. 
However, I think we do not need any of this info here. We are covering them in the Plugins section already.

Comment (camilamacedo86) [reply docs/book/src/plugins/available-plugins.md:10]: But I think you are trying to add the content of : https://github.com/kubernetes-sigs/kubebuilder/blob/master/VERSIONING.md#versioning ( wdyt about a new section versioning only for it? hen, we can add all from this doc add clean up it as well to keep the info centralized in a place?

File: docs/book/src/plugins/available-plugins.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Available plugins
+
+This section describes the plugins supported and shipped in with the Kubebuilder project see: 
+- [go/v3](go-v3-plugin.md)
+- [go/v2](go-v2-plugin.md)
```
Comment (camilamacedo86) [inline docs/book/src/plugins/available-plugins.md:5]: I think we need to have the plugins as sub-sections.

Note that this section will describe any plugin that is shipped by default via KB. (no only theese ones)

File: docs/book/src/plugins/go-v2-plugin.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# go/v2 plugin
+
+This plugin is also used to build operators similar to [go/v3](go-v3-plugin.md). Check the following [example](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v2) to view the files created or updated by this plugin when used with the supported subcommands.
+
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:4]:
```suggestion
The go/v2 plugins have the purpose to scaffold Golang projects and keep the backwards compatibility with the default scaffold made using kubebuilder CLI 2.x.z releases.   

<node>
You can check samples using this plugin by looking at the `project-v2-<options>` samples under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
</node>  

```

File: docs/book/src/plugins/go-v2-plugin.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+## How to use it ?
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:5]:
```suggestion
## When should I use this plugin

Only if you are looking to scaffold a project with the legacy layout. Otherwise, it is recommended you to use the default Golang version plugin. 

<aside class="note warning">

Be aware that this plugin version does not provide a scaffold compatible with the latest versions of the dependencies used in order to keep its backwards compatibility. 

</aside>


## How to use this plugin ?
```

File: docs/book/src/plugins/go-v2-plugin.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+This plugin should only be used to scaffold Go projects which are not backwards compatible with the Kubebuilder layout >= 2.0.0 and < 3.0.0. 
+Use the following command to create a new project with the go/v2 plugin:
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:7]:
```suggestion
To initialize a Golang project using the legacy layout and with this plugin run: 
```

File: docs/book/src/plugins/go-v2-plugin.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+```sh
+kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project --plugins=go/v2
+```
+All the other subcommands supported by go/v2 plugin can be executed similarly.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:12]:
```suggestion
<aside>

By creating a project with this plugin, the PROJECT file scaffold will be using the previous schema (_project version 2_).  So that Kubebuilder CLI knows what plugin version was used and will call its subcommands such as `create api` and `create webhooks`.  Note that further Golang plugins versions use the new Project file schema, which tracks the information about what plugins and versions have been used so far. 

</aside>
```

File: docs/book/src/plugins/go-v2-plugin.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+## Subcommands supported by the plugin
+- [x] Init
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:15]:
```suggestion
- Init ($ kubebuilder init [OPTIONS])
```

I think we do not need the x. Also, could we supplement the follow ones? WDYT?

File: docs/book/src/plugins/go-v3-plugin.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# go/v3
+
+This is the default plugin that is used by Kubebuilder to create operators. It basically scaffolds all the boilerplate code required to create and design operators. Check the following [example](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3) to view the files created or updated by this plugin when used with the supported subcommands. You can also follow the [Building CronJob](cronjob-tutorial.md) tutorial to get a step-by-step explanation of how the plugin works with different subcommands.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:3]: See:

All Operators are made of [controllers](https://github.com/kubernetes-sigs/controller-runtime), but not all sets of controllers are operators. Operators are sets of controllers that encode human operational knowledge for a specific application (e.g. the Prometheus Operator). Controllers are more general -- for instance, the namespace service account controller isn't an operator because it doesn't have knowledge of a specific application -- it's very general. Kubebuilder helps constructs sets of controllers. Some of those are Operators, some might not be.

So, could we update it accordingly? WDYT?

**Review by camilamacedo86 — COMMENTED** [review]
Great work 🥇 

Just a few nits commented in the PR
Also, see that we need to squash the commits to get this one merged.
Could you please to do that?

File: docs/book/src/plugins/available-plugins.md (hunk 1 above)
Comment (Ashwin901) [reply docs/book/src/plugins/available-plugins.md:10]: Sorry for the late response. So should I create a new subsection under plugins section with the name plugins versioning?

File: docs/book/src/plugins/go-v2-plugin.md (hunk 7 above)
Comment (Ashwin901) [reply docs/book/src/plugins/go-v2-plugin.md:19]: Ya I think that would be better. I will make the changes

File: docs/book/src/plugins/go-v3-plugin.md (hunk 8 above)
Comment (Ashwin901) [reply docs/book/src/plugins/go-v3-plugin.md:3]: Got it. I will make the changes accordingly

File: docs/book/src/plugins/available-plugins.md (hunk 1 above)
Comment (camilamacedo86) [reply docs/book/src/plugins/available-plugins.md]: I think that would be great 👍

File: docs/book/src/plugins/available-plugins.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# Available plugins
+
+This section describes the plugins supported and shipped in with the Kubebuilder project.
+
+But any user can create their own plugins :
+- [Creating your own plugins](creating-plugins.md). 
+- [Plugins Phase 2](https://github.com/kubernetes-sigs/kubebuilder/issues/1378) - will allow users to use these external options as helpers to perform the scaffolds with the tool.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/available-plugins.md:7]: WDYT about it be a note? Would not render better?

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Seems to be missing address the comments + rebase with master to solve the conflict with: https://github.com/kubernetes-sigs/kubebuilder/pull/2562

Then we can get this one merged 🥇

File: docs/book/src/plugins/available-plugins.md (hunk 1 above)
Comment (Ashwin901) [reply docs/book/src/plugins/available-plugins.md]: Okay

**Comment by Ashwin901** [comment]
Okay. I will do the changes and push it asap

**Comment by Ashwin901** [comment]
Hey @camilamacedo86 I made the necessary changes and also rebased with master. Can you let me know what the issue is?

File: docs/book/src/plugins/declarative-v1.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
+# Declarative Plugin
```
Comment (camilamacedo86) [inline docs/book/src/plugins/declarative-v1.md:1]: Shows to be missing be rebased with master and squash all commits
See that this one should not be part of this PR.

File: docs/book/src/plugins/declarative-v1.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+The following scaffolds will be created or updated by this plugin:
+
+* `controllers/*_controller.go`
+* `api/*_types.go`
+* `channels/packages/<packagename>/<version>/manifest.yaml`
+* `channels/stable`
+* `Dockerfile`
+
+## Further ressources
```
Comment (rashmigottipati) [inline docs/book/src/plugins/declarative-v1.md:51]:
```suggestion
## Further resources
```

File: docs/book/src/plugins/available-plugins.md (hunk 9 above)
Comment (rashmigottipati) [reply docs/book/src/plugins/available-plugins.md:9]: I would also like to point to point to to the phase 2 design document we have as that would be more accurate than the details in feature request regarding how we will create and use external plugins. Here's the [link](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-2.md) to the phase 2 external plugins design document.

File: docs/book/src/plugins/declarative-v1.md (hunk 10 above)
Comment (Ashwin901) [reply docs/book/src/plugins/declarative-v1.md:1]: I will fix this asap

**Comment by Ashwin901** [comment]
Hey @camilamacedo86 there was some issue with rebase and I unknowingly closed the pull request. But everything seems to be fine now. Can u please review this?

File: docs/book/src/plugins/available-plugins.md [hunk 12]
```diff
@@ -1,5 +1,11 @@
-# Available Plugins
+# Available plugins
 
-This section details how to use the currently available plugins in Kubebuilder. 
+This section describes the plugins supported and shipped in with the Kubebuilder project.
 
-  - [Declarative V1](declarative-v1.md)
```
Comment (camilamacedo86) [inline docs/book/src/plugins/available-plugins.md:5]: You are removing things from the master branch that you should not
You probably missed it up with the rebase 
You can compare your local branch with master and ensure that you have only what is required to have

File: docs/book/src/plugins/available-plugins.md [hunk 13]
```diff
@@ -1,5 +1,11 @@
...
\ No newline at end of file
+<aside class="note">
+
+But any user can create their own plugins see:
```
Comment (camilamacedo86) [inline docs/book/src/plugins/available-plugins.md:7]:
```suggestion
<h1>But you can create your own plugins, see:</h1>
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
+# go/v3
+
+This is the default plugin that is used by Kubebuilder to construct sets of [controllers](https://github.com/kubernetes-sigs/controller-runtime). It basically scaffolds all the boilerplate code required to create and design controllers. Check the following [example](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3) to view the files created or updated by this plugin when used with the supported subcommands. You can also follow the [Building CronJob](cronjob-tutorial.md) tutorial to get a step-by-step explanation of how the plugin works with different subcommands.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:3]:
```suggestion
Kubebuilder tool will scaffold the go/v3 plugin by default. This plugin is a composition of the plugins ` kustomize.common.kubebuilder.io/v1` and `base.go.kubebuilder.io/v3`. By using you can scaffold the default project which is a helper to construct sets of [controllers](https://github.com/kubernetes-sigs/controller-runtime). 

It basically scaffolds all the boilerplate code required to create and design controllers. Note that by following the [quickstart][quickstart] you will be using this plugin. 

<aside class="note">
<h1>Examples</h1>

Samples are provided under the [testdata][testdata] directory of the Kubebuilder project. You can check the go/v3 content by looking at all projects which are named with `v3`. Check the project-v3 to see the common based without the additional options. 

</aside>

Comment (camilamacedo86) [reply docs/book/src/plugins/go-v3-plugin.md:3]: We also need add the alias [link]: link at the bottom 
See the doc added for the declarative plugin already.

File: docs/book/src/plugins/go-v3-plugin.md [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:4]:
```suggestion
## When to use it

- If you are looking to scaffold Golang projects to develop projects using [controllers](https://github.com/kubernetes-sigs/controller-runtime)

File: docs/book/src/plugins/go-v3-plugin.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+## How to use it ?
+As go/v3 is the default plugin there is no need to explicitly mention to Kubebuilder to use this plugin. 
+To create a new project with the go/v3 plugin the following command can be used:
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:7]:
```suggestion

To create a new project with the `go/v3` plugin the following command can be used: 
```

Use >`` for numbers type always.

File: docs/book/src/plugins/go-v3-plugin.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+## How to use it ?
+As go/v3 is the default plugin there is no need to explicitly mention to Kubebuilder to use this plugin. 
+To create a new project with the go/v3 plugin the following command can be used: 
+```sh
+kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project
+```
+All the other subcommands supported by the go/v3 plugin can be executed similarly.
+
+If needed we can explicitly tell Kubebuilder to use the go/v3 plugin by adding ```--plugins=go/v3```.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:13]:
```suggestion

<aside class="note">
<h1>Note</h1>

Also, if you need you can explicitly inform the plugin via the option provided `--plugins=go/v3`.

</aside> 
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+## Subcommands supported by the plugin
+-  Init -  `kubebuilder init [OPTIONS]`
+-  Edit -  `kubebuilder edit [OPTIONS]`
+-  Create API -  `kubebuilder create api [OPTIONS]`
+-  Create Webhook - `kubebuilder create webhook [OPTIONS]`
+
+## Further resources
+- For full implementation of this plugin check out [go/v3](https://pkg.go.dev/sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3).
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:22]: I liked that we added the go docs. But it is not the code implementation.
So, I'd recommend something like :  

- Check in the [main.go](https://github.com/kubernetes-sigs/kubebuilder/blob/master/cmd/main.go) how the plugin is composited
- Check the code implementation for the base.go.kubebuilder.io/v3
- Check the code implementation for  kustomize.common.kubebuilder.io/v1
- Check the [controller-runtime] to know more about controllers

File: docs/book/src/plugins/plugins-versioning.md [hunk 19]
```diff
@@ -0,0 +1,23 @@
+# Plugins Versioning
+Plugin version represents the version of an individual plugin, as well as the corresponding scaffolding that it generates. This version is defined in a plugin key, ex. `go.kubebuilder.io/v2`. See the [design doc](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/plugins/extending-cli.md#plugin-versioning) for more details.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins-versioning.md:2]:
```suggestion

Plugin version represents the version of an individual plugin, as well as the corresponding scaffolding that it generates. This version is defined in a plugin key, ex. `go.kubebuilder.io/v2`. See the [design doc](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/plugins/extending-cli.md#plugin-versioning) for more details.
```

File: docs/book/src/plugins/plugins-versioning.md [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+## Incrementing plugin versions
+Changes made to plugins only require a plugin version increase if and only if a change is made to a plugin
```
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins-versioning.md:5]:
```suggestion

Changes made to plugins only require a plugin version increase if and only if a change is made to a plugin
```

File: docs/book/src/plugins/plugins-versioning.md [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+for users to upgrade their projects from `vX` to `v(X+1)-alpha`.
+
+<aside class="note">
+<h1>Example</h1>
+
+Kubebuilder scaffolds projects with plugin `go.kubebuilder.io/v3` by default.
+
+You create a feature that adds a new marker to the file `main.go` scaffolded by `init` that `create api` will use to update that file. The changes introduced in your feature would cause errors if used with projects built with plugins `go.kubebuilder.io/v2` without users manually updating their projects. Thus, your changes introduce a breaking change to plugin `go.kubebuilder.io`, and can only be merged into plugin version `v3-alpha`. This plugin's package should exist already.
+
+</aside>
```
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins-versioning.md:23]: We need +1 section : https://github.com/kubernetes-sigs/kubebuilder/blob/master/VERSIONING.md#versioning

File: docs/book/src/plugins/plugins-versioning.md [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+that breaks projects scaffolded with the previous plugin version. Once a plugin version `vX` is stabilized (it doesn't
+have an "alpha" or "beta" suffix), a new plugin package should be created containing a new plugin with version
+`v(X+1)-alpha`. Typically this is done by (semantically) `cp -r pkg/plugins/golang/vX pkg/plugins/golang/v(X+1)` then updating
+version numbers and paths. All further breaking changes to the plugin should be made in this package; the `vX`
+plugin would then be frozen to breaking changes.
+
+You must also add a migration guide to the [migrations](../migrations.md)
+section of the Kubebuilder book in your PR. It should detail the steps required
+for users to upgrade their projects from `vX` to `v(X+1)-alpha`.
+
```
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins-versioning.md:15]: It shows missed up we have 2 sections there Introducing changes to plugins and Incrementing versions

File: docs/book/src/SUMMARY.md [hunk 23]
```diff
@@ -109,7 +109,10 @@
   - [Extending the CLI](./plugins/extending-cli.md)
   - [Creating your own plugins](./plugins/creating-plugins.md)
   - [Available Plugins](./plugins/available-plugins.md)
+    - [go/v3 plugin](./plugins/go-v3-plugin.md)
```
Comment (camilamacedo86) [inline docs/book/src/SUMMARY.md:112]: This are the guidelines that we should follow up: https://kubernetes.io/docs/contribute/style/style-guide/
If you would like would be great we create a PR to add this one to the contribution guide as well.

File: docs/book/src/plugins/plugins-versioning.md (hunk 22 above)
Comment (Ashwin901) [reply docs/book/src/plugins/plugins-versioning.md:15]: Sure. I will will another raise for plugin versioning

File: docs/book/src/SUMMARY.md (hunk 23 above)
Comment (Ashwin901) [reply docs/book/src/SUMMARY.md:112]: Hey @camilamacedo86 I didn't  quite understand this comment. Can you please explain it once again?

Comment (Ashwin901) [reply docs/book/src/SUMMARY.md:112]: I have raised a PR

File: docs/book/src/plugins/plugins-versioning.md (hunk 22 above)
Comment (Ashwin901) [reply docs/book/src/plugins/plugins-versioning.md]: I have raise a PR #2588

File: docs/book/src/SUMMARY.md (hunk 23 above)
Comment (camilamacedo86) [reply docs/book/src/SUMMARY.md:112]: Sure. It is only for you know.
We ought to follow the K8s style for documentation which is defined in this link

I thought that we did have not this info in our Contribution Guide: https://github.com/kubernetes-sigs/kubebuilder/blob/master/CONTRIBUTING.md . However I just checked that, yes we have: (which is great)

> There are certain writing [style guide](https://kubernetes.io/docs/contribute/style/style-guide/)lines for Kubernetes documentation, the checkout style guide for more information.

File: docs/book/src/plugins/go-v2-plugin.md [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+</aside>
+
+## Subcommands supported by the plugin
+
+-  Init -  `kubebuilder init [OPTIONS]`
+-  Edit -  `kubebuilder edit [OPTIONS]`
+-  Create API -  `kubebuilder create api [OPTIONS]`
+-  Create Webhook - `kubebuilder create webhook [OPTIONS]`
+
+
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:41]:
```suggestion
```
nit unnecessary space

File: docs/book/src/plugins/go-v3-plugin.md [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+## Subcommands supported by the plugin
+
+-  Init -  `kubebuilder init [OPTIONS]`
+-  Edit -  `kubebuilder edit [OPTIONS]`
+-  Create API -  `kubebuilder create api [OPTIONS]`
+-  Create Webhook - `kubebuilder create webhook [OPTIONS]`
+
+## Further resources
+
+- To check how plugins are composited read [main.go][plugins-main].
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:47]:
```suggestion
- To check how plugins are composited by looking at this definition in the [main.go][plugins-main].
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 26]
```diff
@@ -0,0 +1,57 @@
...
+- Check the code implementation of [go/v3 plugin][v3-plugin].
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:48]:
```suggestion
- Check the code implementation of the [base Golang plugin `base.go.kubebuilder.io/v3`][v3-plugin].
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+- Check the code implementation of [kustomize plugin][kustomize-plugin].
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:49]:
```suggestion
- Check the code implementation of [Kustomize/v1 plugin][kustomize-plugin].
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+- Check [controller-runtime] to know more about controllers.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:50]:
```suggestion
- Check [controller-runtime][controller-runtime] to know more about controllers.
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 29]
```diff
@@ -0,0 +1,57 @@
...
+
+Kubebuilder tool will scaffold the go/v3 plugin by default. This plugin is a composition of the plugins ` kustomize.common.kubebuilder.io/v1` and `base.go.kubebuilder.io/v3`. By using you can scaffold the default project which is a helper to construct sets of [controllers][controller-runtime]. 
+
+It basically scaffolds all the boilerplate code required to create <b>and</b> design controllers. Note that by following the [quickstart][quickstart] you will be using this plugin. 
+
+<aside class="note">
+
+<h1>Examples</h1>
+
+Samples are provided under the [testdata][testdata] directory of the Kubebuilder project. You can check the go/v3 content by looking at all projects which are named with `v3`. Check the project-v3 to see the common based without the additional options.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:11]:
```suggestion
Samples are provided under the [testdata][testdata] directory of the Kubebuilder project. You can check samples using this plugin by looking at the `project-v3-<options>` under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
```

File: docs/book/src/plugins/go-v2-plugin.md [hunk 30]
```diff
@@ -0,0 +1,48 @@
+# go/v2 plugin
+
+The `go/v2` plugin has the purpose to scaffold Golang projects to help users to build projects with [controllers][controller-runtime] and keep the backwards compatibility with the default scaffold made using Kubebuilder CLI `2.x.z` releases.   
+
+<node>
+
+You can check samples using this plugin by looking at the `project-v2-<options>` samples under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:7]:
```suggestion
You can check samples using this plugin by looking at the `project-v2-<options>` directories under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
```

**Review by camilamacedo86 — COMMENTED** [review]
I did another review round. 
I think are closer now to getting this good enough to be merged

Great work !

File: docs/book/src/SUMMARY.md (hunk 23 above)
Comment (Ashwin901) [reply docs/book/src/SUMMARY.md:112]: Okay

File: docs/book/src/plugins/go-v3-plugin.md [hunk 31]
```diff
@@ -0,0 +1,57 @@
+# go/v3
+
+Kubebuilder tool will scaffold the go/v3 plugin by default. This plugin is a composition of the plugins ` kustomize.common.kubebuilder.io/v1` and `base.go.kubebuilder.io/v3`. By using you can scaffold the default project which is a helper to construct sets of [controllers][controller-runtime]. 
+
+It basically scaffolds all the boilerplate code required to create <b>and</b> design controllers. Note that by following the [quickstart][quickstart] you will be using this plugin.
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:5]: Why do we need the bold?

File: docs/book/src/plugins/go-v3-plugin.md [hunk 32]
```diff
@@ -0,0 +1,57 @@
...
+
+<h1>Examples</h1>
+
+Samples are provided under the [testdata][testdata] directory of the Kubebuilder project. You can check samples using this plugin by looking at the `project-v3-<options>` projects under the [testdata][testdata] directory on the root directory of the Kubebuilder project.
+
+</aside>
+
+## When to use it
+
+- <b>If</b> you are looking to scaffold Golang projects to develop projects using [controllers][controller-runtime]
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v3-plugin.md:17]: Why do we need the bold?

File: docs/book/src/plugins/go-v2-plugin.md [hunk 33]
```diff
@@ -0,0 +1,47 @@
...
+
+## How to use it ?
+
+To initialize a Golang project using the legacy layout and with this plugin run, e.g.:
+
+```sh
+kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project --plugins=go/v2
+```
+<aside class="note">
+
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:29]:
```suggestion
<h1> Note </h1>
```
We meed to add the title for it render right

File: docs/book/src/plugins/go-v2-plugin.md [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+You can check samples using this plugin by looking at the `project-v2-<options>` directories under the [testdata][testdata] projects on the root directory of the Kubebuilder project.
+
+</node>  
+
+## When should I use this plugin
+
+Only if you are looking to scaffold a project with the legacy layout. Otherwise, it is recommended you to use the default Golang version plugin. 
+
+<aside class="note warning">
+
```
Comment (camilamacedo86) [inline docs/book/src/plugins/go-v2-plugin.md:16]:
```suggestion
<h1> Supportability </h1>
```

**Review by camilamacedo86 — APPROVED** [review]
For me, just missing a few nits. 
Remove the bold and ensure that the notes has a title

Otherwise, 
/approved

File: docs/book/src/plugins/go-v3-plugin.md (hunk 31 above)
Comment (Ashwin901) [reply docs/book/src/plugins/go-v3-plugin.md:5]: Oh sorry. I was checking something and forgot to remove it. Will push the changes

**Comment by Ashwin901** [comment]
@camilamacedo86 is everything fine now? Or are there any pending changes?

**Comment by camilamacedo86** [comment]
Hi @Ashwin901,

> @camilamacedo86 is everything fine now? Or are there any pending changes?

It shows all fine for me. But I pinged in the channel to see if we get help on the review.
IMO after we merge all docs we will need to work to just shape all since we split we might need to tackle a little.

File: docs/book/src/plugins/available-plugins.md [hunk 35]
```diff
@@ -1,5 +1,15 @@
...
\ No newline at end of file
+<aside class="note">
+
+<h1>But you can create your own plugins, see:</h1>
+
+- [Creating your own plugins][create-plugins]. 
+- [Plugins Phase 2][plugins-phase-2] - will allow users to use these external options as helpers to perform the scaffolds with the tool.
+
+</aside>
+
```
Comment (AlmogBaku) [inline docs/book/src/plugins/available-plugins.md:13]: missing a link to the v3

File: docs/book/src/plugins/available-plugins.md [hunk 36]
```diff
@@ -1,5 +1,15 @@
...
+[create-plugins]: creating-plugins.md
```
Comment (AlmogBaku) [inline docs/book/src/plugins/available-plugins.md:14]: this line should probably be inside of the `aside`?

File: docs/book/src/plugins/available-plugins.md [hunk 37]
```diff
@@ -1,5 +1,15 @@
...
+[plugins-phase-2]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-2.md
```
Comment (AlmogBaku) [inline docs/book/src/plugins/available-plugins.md:15]: this line should probably be inside of the `aside`?

File: docs/book/src/plugins/go-v3-plugin.md [hunk 38]
```diff
@@ -0,0 +1,57 @@
+# go/v3
+
+Kubebuilder tool will scaffold the go/v3 plugin by default. This plugin is a composition of the plugins ` kustomize.common.kubebuilder.io/v1` and `base.go.kubebuilder.io/v3`. By using you can scaffold the default project which is a helper to construct sets of [controllers][controller-runtime].
```
Comment (AlmogBaku) [inline docs/book/src/plugins/go-v3-plugin.md:3]: "Kubebuilder tool will scaffold the go/v3 plugin by default." ->  add a new paragraph after this line, and perhaps make it bold.
"By using ..." -> "Using the plugin, you can..."

File: docs/book/src/plugins/go-v2-plugin.md [hunk 39]
```diff
@@ -0,0 +1,51 @@
+# go/v2 plugin
+
+The `go/v2` plugin has the purpose to scaffold Golang projects to help users to build projects with [controllers][controller-runtime] and keep the backwards compatibility with the default scaffold made using Kubebuilder CLI `2.x.z` releases.
```
Comment (AlmogBaku) [inline docs/book/src/plugins/go-v2-plugin.md:3]: "...to scaffold Golang projects to help users..." -> "... to scaffold Golang projects by helping users..."

Comment (AlmogBaku) [reply docs/book/src/plugins/go-v2-plugin.md:3]: "...to scaffold..." -> "...of scaffolding..."

File: docs/book/src/plugins/go-v2-plugin.md [hunk 40]
```diff
@@ -0,0 +1,51 @@
...
+
+The `go/v2` plugin has the purpose to scaffold Golang projects to help users to build projects with [controllers][controller-runtime] and keep the backwards compatibility with the default scaffold made using Kubebuilder CLI `2.x.z` releases.   
+
+<node>
+
+You can check samples using this plugin by looking at the `project-v2-<options>` directories under the [testdata][testdata] projects on the root directory of the Kubebuilder project.
+
+</node>  
+
+## When should I use this plugin
```
Comment (AlmogBaku) [inline docs/book/src/plugins/go-v2-plugin.md:11]: add a question mark

**Review by AlmogBaku — CHANGES_REQUESTED** [review]

**Comment by AlmogBaku** [comment]
Thank you for your contribution @Ashwin901 !
I suggest a few changes in the comments :)

File: docs/book/src/plugins/available-plugins.md [hunk 41]
```diff
@@ -1,5 +1,15 @@
...
+# Available plugins
 
-This section details how to use the currently available plugins in Kubebuilder. 
+This section describes the plugins supported and shipped in with the Kubebuilder project.
 
-  - [Declarative V1](declarative-v1.md)
\ No newline at end of file
+<aside class="note">
+
+<h1>But you can create your own plugins, see:</h1>
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply docs/book/src/plugins/available-plugins.md:7]: Hun, the page speaks about available plugins. then, could we not consider this content a note, or observation? Because of this in POV it shows fit better in the <aside>

File: docs/book/src/plugins/available-plugins.md [hunk 42]
```diff
@@ -1,5 +1,15 @@
...
+
+- [Creating your own plugins][create-plugins]. 
+- [Plugins Phase 2][plugins-phase-2] - will allow users to use these external options as helpers to perform the scaffolds with the tool.
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply docs/book/src/plugins/available-plugins.md:10]: same ^

File: docs/book/src/plugins/available-plugins.md (hunk 36 above)
Comment (camilamacedo86) [reply docs/book/src/plugins/available-plugins.md:14]: Nops, the link alias should be at the bottom of the docs pages to make it easier we keep the maintained. 
That is the standard.

File: docs/book/src/plugins/available-plugins.md (hunk 37 above)
Comment (camilamacedo86) [reply docs/book/src/plugins/available-plugins.md:15]: same ^

File: docs/book/src/plugins/available-plugins.md (hunk 42 above)
[the remark this answers is not in the document]
Comment (Ashwin901) [reply docs/book/src/plugins/available-plugins.md:10]: So the two items should be placed outside the aside section?

**Comment by camilamacedo86** [comment]
Hi @Ashwin901,

mainly all suggestions are addressed and it shows in a good shape to get merged.
We still need to add. in the questions, however, I think the others plugins docs have done so far also need it. 

So, @AlmogBaku @Ashwin901 

Let's get this merged to not lost the contribution and work on some improvements shapes required as follows. I hope that is fine and I hope that you can help with this follow up so we are able to organize and improve the whole content.  

/lgtm
